.github/conftest-unified-ci-rules.rego

```diff
@@ -88,6 +88,35 @@ deny[msg] {
         [concat(", ", get_jobs_after_checkresults_without_ifalways(input.jobs))])
 }
 
+deny[msg] {
+    # only enforced on Unified CI since it is specific to utils-flaky-tests-summary job
+    input.name == "CI"
+    input.jobs["utils-flaky-tests-summary"]
+
+    jobs_with_flaky_outputs := get_jobs_with_flaky_outputs(input.jobs)
+    flaky_summary_needs := {need | need := input.jobs["utils-flaky-tests-summary"].needs[_]}
+
+    missing_jobs := jobs_with_flaky_outputs - flaky_summary_needs
+    count(missing_jobs) > 0
+
+    msg := sprintf("The utils-flaky-tests-summary job is missing dependencies on jobs with flakyTests outputs! Missing job IDs: %s",
+        [concat(", ", missing_jobs)])
+}
+
+deny[msg] {
+    # only enforced on Unified CI since it is specific to utils-flaky-tests-summary job
+    input.name == "CI"
+    input.jobs["utils-flaky-tests-summary"]
+
+    needs_list := input.jobs["utils-flaky-tests-summary"].needs
+    sorted_needs := sort(needs_list)
+
+    needs_list != sorted_needs
```

Comment:
Nice, I would've been ok with just a comment and setting an example by having the list ordered after your PR but this is even bettter :)